rclrs/src/context.rs

```diff
@@ -118,6 +127,47 @@ impl Context {
     }
 }
 
+/// Returns non-ROS arguments held by context.
+///
+/// This function should be called after rcl_init. `rcl_arguments` should be `global_arguments`
+/// field of initialized context. `args` is array of input arguments passed to node.
+fn get_non_ros_arguments(
+    rcl_arguments: *const rcl_arguments_t,
+    args: &[*const c_char],
+) -> Result<Vec<String>, RclrsError> {
+    // SAFETY: In case of error -1 will be returned and handled a line below.
+    let args_status = unsafe { rcl_arguments_get_count_unparsed(rcl_arguments) };
+    if args_status == -1 {
+        return Err(RclrsError::RclError {
+            code: InvalidArgument,
+            msg: None,
+        });
+    }
```

Comment:
```suggestion
    let args_status = unsafe { rcl_arguments_get_count_unparsed(rcl_arguments).ok()? };
```

Reply:
`rcl_arguments_get_count_unparsed` does not return `Result` but  `::std::os::raw::c_int` as I can see in generated bindings:
```rs
pub fn rcl_arguments_get_count_unparsed(args: *const rcl_arguments_t) -> ::std::os::raw::c_int;
```
Could you please explain this suggestion?

Also, suggested change does not compile:
```
error[E0308]: mismatched types
   --> src/context.rs:140:23
    |
140 |     if args_status == -1 {
    |                       ^^ expected `()`, found integer

error[E0605]: non-primitive cast: `()` as `usize`
   --> src/context.rs:147:20
    |
147 |     let args_num = args_status as usize;
    |                    ^^^^^^^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
```

Reply:
I think this should be

```
let args_count = unsafe { rcl_arguments_get_count_unparsed(rcl_arguments) };
debug_assert!(args_count != -1);
```

See also https://github.com/ros2-rust/ros2_rust/blob/main/docs/CONTRIBUTING.md#return-type-to-use-when-wrapping-rcl-functions.